src/gpu/intel/jit/gemm/generator/pieces/common.cxx

```diff
@@ -449,15 +446,17 @@ GRFRange BLASKernelGenerator<hw>::loadVector(Type Tsrc, Type Tdst, Subregister p
 
 // Set a matrix to zero.
 template <HW hw>
-void BLASKernelGenerator<hw>::zeroMatrix(const GRFMultirange &r, const CommonStrategy &strategy)
+void Generator<hw>::zeroMatrix(const GRFMultirange &r, const CommonStrategy &strategy)
 {
+    int i = 0;
     map<uint32_t>(hw, r, r, strategy, [&](int esize, GRF reg, GRF _) {
-        mov(esize, reg, uint16_t(0));
+        (i++ & 1) ? mov(esize, reg.f(), 0.f)
+                  : mov(esize, reg, 0);
```

Comment:
Ah, there it is. I'm wondering why onednn haven't use this dual pipeline zero strategy :)

Reply:
Actually, we _should_ be peeling off the first mad/dp4a/dpas and using a null src to avoid this zeroing entirely. Just hasn't been done...